### PR TITLE
Rename username to userId

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
To clarify and have it in sync across all projects:
userId: internally id, needed for e.g. dav/trashbin/$userId/

loginName: used for authenticate on every login/network connection, determinated by first login (weblogin/old login): can be anything: email, name, name with withespaces

displayName: Name chosen by user, only displayed, no usage at all

There will be soon a matching PR in library/files

Signed-off-by: tobiasKaminsky <tobias@kaminsky.me>

### CODE REVIEW

File: src/main/java/com/nextcloud/android/sso/Constants.java [hunk 1]
```diff
@@ -22,7 +22,7 @@
 public class Constants {
 
     // Authenticator related constants
-    public static final String SSO_USERNAME = "username";
+    public static final String SSO_USER_ID = "user_id";
```
Comment (David-Development) [inline src/main/java/com/nextcloud/android/sso/Constants.java]: Changing this line breaks backward compatibility, right? Is there another option to make the transition soft? And we would have to upgrade third party apps first before you release this into the nextcloud files app.

Maybe add something like this: (in the `AccountImporter.java`)

```java
String userId = future.getString(Constants.SSO_USER_ID);
if (userId == null) {
  // backwards compatibility
  userId = future.getString("username");
}
```

Comment (tobiasKaminsky) [reply src/main/java/com/nextcloud/android/sso/Constants.java]: This is a nice idea.

**Comment by David-Development** [comment]
@tobiasKaminsky is this ready to be merged?

**Comment by tobiasKaminsky** [comment]
Yes, as this ensures that it backward compatible, we can merge this any time
(corresponding PR on files app will also be merged soon)

**Pull request merged**